pkg/util/runtime/runtime.go

```diff
@@ -23,14 +23,22 @@ import (
 )
 
 // For testing, bypass HandleCrash.
+// TODO: should no longer be necessary now that HandleCrash doesn't eat panics.
 var ReallyCrash bool
 
 // PanicHandlers is a list of functions which will be invoked when a panic happens.
 var PanicHandlers = []func(interface{}){logPanic}
 
-//TODO search the public functions
-// HandleCrash simply catches a crash and logs an error. Meant to be called via defer.
-// Additional context-specific handlers can be provided, and will be called in case of panic
+// HandleCrash simply catches a crash and logs an error. Meant to be called via
+// defer.  Additional context-specific handlers can be provided, and will be
+// called in case of panic.  HandleCrash actually crashes, after calling the
+// handlers and logging the panic message.
+//
+// TODO: remove this function. We are switching to a world where it's safe for
+// apiserver to panic, since it will be restarted by kubelet. At the beginning
+// of the Kubernetes project, nothing was going to restart apiserver and so
+// catching panics was important. But it's actually much simpler for montoring
+// software if we just exit when an unexpected panic happens.
 func HandleCrash(additionalHandlers ...func(interface{})) {
```

Comment:
Well, we want the handlers to be called.  The handlers allow us to report to external systems (on my long delayed list is to upstream the sentry code that reports all panics to a remote system).